Update docs

ordabayevy opened the pull request:
- Reformat docstrings to have argument description start on a new line
- Add args to the docstring of `DistributedAnnDataCollectionDataModule`
- Add missing `obs_columns` to the init of `DistributedAnnDataCollectionDataModule`
- Display documented `:special-members:` like `__iter__` in the docs

fedorgrab commented on cellarium/ml/train/training_plan.py:
```diff
@@ -13,16 +13,22 @@
+        module:
+            A ``cellarium.ml`` module to train.
+        optim_fn:
+            A Pytorch optimizer class, e.g., :class:`~torch.optim.Adam`. If ``None``,
```
Out of curiosity, what does Tilda do in this context?

ordabayevy replied:
Tilde means to hide prefix in the class name and only display `Adam` in the docs.

fedorgrab reviewed (changes requested):
Most of my comments pertain to cosmetics and such, so feel free to resolve the discussions if you think otherwise.

ordabayevy commented:
Thanks for the feedback @fedorgrab ! I addressed your comments and added further changes:
- Use `sphinx-autodoc-typehints` to add type hints and defaults to the doc. Removed all manually added default values.
- Added `sphinx-copybutton` extension
- Updated html links to be displayed properly
- Updated examples docstrings to display all command line args on a new line
